tools/cli/domainCommands.go

```diff
@@ -807,6 +831,109 @@ func (d *domainCLIImpl) migrationDomainWorkFlowCheck(c *cli.Context) DomainMigra
 	}
 }
 
+func (d *domainCLIImpl) migrationDynamicConfigCheck(c *cli.Context) DomainMigrationRow {
+	check := true
+
+	resp := dynamicconfig.ListAllProductionKeys()
+
+	var currResps *types.GetDynamicConfigResponse
+	var newResps *types.GetDynamicConfigResponse
+
+	for _, configName := range resp {
+		// Check if the key only has a domain filter
+		if len(configName.Filters()) == 1 && configName.Filters()[0] == dc.DomainName {
+			// Use the helper method to get the GetDynamicConfigRequest
+			currRequest := dynamicconfig.ToGetDynamicConfigFilterRequest("", []dynamicconfig.FilterOption{
+				dynamicconfig.DomainFilter(FlagDomain),
+			})
+
+			newRequest := dynamicconfig.ToGetDynamicConfigFilterRequest("", []dynamicconfig.FilterOption{
+				dynamicconfig.DomainFilter(FlagDestinationDomain),
+			})
+
+			currResp, err := d.frontendAdminClient.GetDynamicConfig(context.Background(), currRequest)
+			if err != nil {
+				ErrorAndExit("Failed to fetch dynamic config for current domain.", err)
+			}
+			newResp, err := d.destinationAdminClient.GetDynamicConfig(context.Background(), newRequest)
+			if err != nil {
+				ErrorAndExit("Failed to fetch dynamic config for destination domain.", err)
+			}
+
+			if currResp.Value != newResp.Value {
+				check = false
+			}
+			currResps = currResp
```

Comment:
wrong logic. In the end, you need to collect all `MismatchedDynamicConfig` results